### PR TITLE
Add test for bool int array checker and change class inheritance

### PR DESCRIPTION
**Pull request by KeenanCronyn**
In this PR I have added a test for the "TestIfMatching" function that only looks at valid words passed to it.

I had this as a private function, so in order to get access I changed the inheritance of my test class, and also stopped the CheatFunctions class from being a static class.

I'm not sure if this is a mistake, and if I should have done it a different way. It feels like an amateur move, but I'm not sure.

### CODE REVIEW

File: src/WordscapesCheatTests/CheatFunctionsTests.cs [hunk 1]
```diff
@@ -21,22 +22,25 @@ public void BuildOccurenceArray_PassValidString_Succeed(string input, int[] expe
...
-        //    CollectionAssert.AreEqual(expected, CheatFunctions.BuildOccurenceArray(input));
-        //}
+        [DataTestMethod]
+        [DataRow("abc", "cab")]
+        public void TestIfMatching_PassValidArray_Succeed(string givenLetters, string wordInDictionary)
+        {
+            int[] givenLettersArray = CheatFunctions.BuildOccurenceArray(givenLetters);
+            int[] wordArray = CheatFunctions.BuildOccurenceArray(wordInDictionary);
 
+            Assert.IsTrue(CheatFunctions.TestIfMatching(givenLettersArray, wordArray));
```
Comment (KeenanCronyn) [inline src/WordscapesCheatTests/CheatFunctionsTests.cs]: This is my new test

File: src/WordscapesCheatTests/CheatFunctionsTests.cs [hunk 2]
```diff
@@ -21,22 +22,25 @@ public void BuildOccurenceArray_PassValidString_Succeed(string input, int[] expe
         }
 
         [DataTestMethod]
-        [DataRow("abc", new int[] { 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 })]
-        [DataRow("azaza", new int[] { 3, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 2 })]
-        [DataRow("", new int[] { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 })]
-        public void BuildMatchingWordsArray_ValidString_Succeed(string input, string[] expectedResult)
+        [DataRow("abc", "cab")]
+        [DataRow("test", "set;stet;test")]
+        public void BuildMatchingWordsArray_ValidString_Succeed(string input, string stringResult)
```
Comment (KeenanCronyn) [inline src/WordscapesCheatTests/CheatFunctionsTests.cs]: I think I was working off an old version of master, so I just added in the same code as it had before. 

Not sure if it's better for me to revert back to the correct master and then add in this code.

File: src/WordscapesCheat/Properties/AssemblyInfo.cs [hunk 3]
```diff
@@ -34,3 +34,6 @@
 // [assembly: AssemblyVersion("1.0.*")]
 [assembly: AssemblyVersion("1.0.0.0")]
 [assembly: AssemblyFileVersion("1.0.0.0")]
+
+// Allows the test assembly to view internal members of this 
+[assembly: InternalsVisibleTo("WordscapesCheatTests")]
```
Comment (Cronan) [inline src/WordscapesCheat/Properties/AssemblyInfo.cs]: This is the bit that allows internal stuff to be visible to the test assembly

File: src/WordscapesCheat/CheatFunctions.cs [hunk 4]
```diff
@@ -104,7 +104,8 @@ public static string[] BuildMatchingWordsArray(string givenLetters, List<string>
         /// <param name="givenLettersArray"></param>
         /// <param name="wordArray"></param>
         /// <returns></returns>
-        private static bool TestIfMatching(int[] givenLettersArray, int[] wordArray)
+        internal static bool TestIfMatching(int[] givenLettersArray, int[] wordArray)
```
Comment (Cronan) [inline src/WordscapesCheat/CheatFunctions.cs]: I changed this to internal - see below

File: src/WordscapesCheat/Properties/AssemblyInfo.cs (hunk 3 above)
Comment (KeenanCronyn) [reply src/WordscapesCheat/Properties/AssemblyInfo.cs]: Great, that makes a lot of sense

File: src/WordscapesCheat/CheatFunctions.cs (hunk 4 above)
Comment (KeenanCronyn) [reply src/WordscapesCheat/CheatFunctions.cs]: I see. So, internal access modifier allows classes specified in the assembly to access that method, and so you included the class in the assembly. Therefore, any classes declared in the assembly can then be accessed with the right modifier.

**Comment by KeenanCronyn** [comment]
Fantastic, thank you Dad, this makes a lot of sense, thank you.

**Pull request merged**